### PR TITLE
Feat/4250 update components for publications

### PR DESCRIPTION
**Pull request by asf-algoo**
#4250 

## Checkpoints

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)
Already existing tests updated.
- [x] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified
- [x] Original authors of the features included in a multi-feature branch (maintenance fixes -> develop, security fixes -> develop, …) should be part of the reviewers, especially if you encountered merge conflicts.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done

To test:
Test if the timeline and the activity feed still works.

### CODE REVIEW

**Review by Skylsmoi — APPROVED** [review]

File: frontend/src/component/Activity/ActivityList.jsx [hunk 1]
```diff
@@ -28,15 +29,27 @@ const ENTITY_TYPE_COMPONENT_CONSTRUCTOR = new Map([
...
-        <ContentWithPreviewActivity
-          activity={activity}
+        <FeedItemWithPreview
+          breadcrumbsList={breadcrumbsList}
+          commentList={activity.commentList}
+          content={activity.content}
+          eventList={activity.eventList}
           key={activity.id}
+          lastModifier={activity.newestMessage.fields.author}
+          lastModification={coreEventType}
```
Comment (Skylsmoi) [inline frontend/src/component/Activity/ActivityList.jsx:45]: shouldn't this be named lastModificationType?

File: frontend/src/component/FeedItem/FeedItemWithPreview.jsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        onEventClicked={props.onEventClicked}
+        workspaceId={props.workspaceId}
+      />
+      <Preview content={props.content} />
+      <FeedItemFooter
+        commentList={props.commentList}
+        content={props.content}
+        reactionList={props.reactionList}
+      />
+      {props.children}
```
Comment (Skylsmoi) [inline frontend/src/component/FeedItem/FeedItemWithPreview.jsx:30]: You are not using this, is it for later use?

Comment (gbonaspetti) [reply frontend/src/component/FeedItem/FeedItemWithPreview.jsx:30]: Yes, at issues description: _The component ContentWithPreviewActivity_ (a.k.a. FeedItemWithPreview) _should allow children to add the timeline on it_

File: frontend/src/component/Activity/ActivityList.jsx (hunk 1 above)
Comment (gbonaspetti) [reply frontend/src/component/Activity/ActivityList.jsx]: fixed

File: frontend/src/component/FeedItem/FeedItemWithPreview.jsx (hunk 2 above)
Comment (Skylsmoi) [reply frontend/src/component/FeedItem/FeedItemWithPreview.jsx:30]: ok, good

**Review by Skylsmoi — APPROVED** [review]

**Pull request merged**